### ISSUE TITLE
Update featureOverlay when adding features programatically to selection

### ISSUE THREAD

**Issue by JWGmeligMeyling** [openlayers/ol3#5247]
When adding features to a selection programatically, like so:
```js
var selectedFeatures = selectableInteraction.getFeatures();
selectedFeatures.remove(selectedFeature);
selectedFeatures.push(feature);
```

The map is not updated by default. I manage to hack around this by doing:
```js
selectableInteraction.featureOverlay_.getSource().dispatchEvent('change');
```

Is this the correct way of adding features to a selection programatically? Should this change event be emitted by default?

**Comment by ahocevar** [comment]
Is `selectableIneraction` an `ol.interaction.Select`? If so, the above should work without dispatching an event manually. You can try the following on the console in http://openlayers.org/en/v3.15.1/examples/select-features.html:
```js
var features = selectClick.getFeatures();
// Select a feature interactively now.
var feature = features.item(0);
features.remove(feature); // Map is updated.
features.push(feature); // Map is updated again.
```

**Comment by JWGmeligMeyling** [comment]
It is indeed a. Was this fixed in `v3.15.1`? Because I'm indeed using a  `ol.interaction.Select`, but then in `v3.12.1`?

**Comment by ahocevar** [comment]
The same console sequence also works in v3.12.1.
